### PR TITLE
SWC-6618: create list of associated projects for an AR

### PR DESCRIPTION
**Pull request by hallieswan**
Will be used in SWC to display a list of associated projects on each AR page. See SWC changes [here](https://github.com/Sage-Bionetworks/SynapseWebClient/pull/5247).

### CODE REVIEW

File: packages/synapse-react-client/src/components/AccessRequirementRelatedProjectsList/AccessRequirementRelatedProjectsList.stories.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+} satisfies Meta
+export default meta
+
+type Story = StoryObj<typeof meta>
+
+export const ManyProjects: Story = {
+  args: {
+    accessRequirementId: 9603055,
+  },
+}
```
Comment (hallieswan) [inline packages/synapse-react-client/src/components/AccessRequirementRelatedProjectsList/AccessRequirementRelatedProjectsList.stories.ts:16]: ManyProjects

File: packages/synapse-react-client/src/components/AccessRequirementRelatedProjectsList/AccessRequirementRelatedProjectsList.stories.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+export const OneProject: Story = {
+  args: {
+    accessRequirementId: 9605913,
+  },
+}
```
Comment (hallieswan) [inline packages/synapse-react-client/src/components/AccessRequirementRelatedProjectsList/AccessRequirementRelatedProjectsList.stories.ts:22]: OneProject

File: packages/synapse-react-client/src/components/EntityLink.tsx [hunk 3]
```diff
@@ -84,7 +84,9 @@ export const EntityLink = (props: EntityLinkProps) => {
     } else {
       return (
         <p className={className}>
-          <EntityTypeIcon type={type} style={{ marginRight: '6px' }} />
+          {showIcon && (
+            <EntityTypeIcon type={type} style={{ marginRight: '6px' }} />
+          )}
```
Comment (hallieswan) [inline packages/synapse-react-client/src/components/EntityLink.tsx:89]: Respect `showIcon` prop when `link` prop is false.

File: packages/synapse-react-client/src/components/AccessRequirementRelatedProjectsList/AccessRequirementRelatedProjectsList.tsx [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+  }
+
+  const { data, isError, error } =
+    useSearchAccessRequirementsInfinite(searchRequest)
+  const ar = data?.pages.flatMap(page => page.results)[0] ?? undefined
+
+  const nRelatedProjects = ar?.relatedProjectIds.length || 0
+  const relatedProjectsTitle = `${nRelatedProjects} project${
+    nRelatedProjects === 1 ? '' : 's'
+  }${nRelatedProjects > 0 ? ':' : ''}`
```
Comment (nickgros) [inline packages/synapse-react-client/src/components/AccessRequirementRelatedProjectsList/AccessRequirementRelatedProjectsList.tsx:25]: We already have `pluralize` as a dependency, so you can just use that.

File: packages/synapse-react-client/src/components/AccessRequirementRelatedProjectsList/AccessRequirementRelatedProjectsList.stories.ts (hunk 1 above)
Comment (nickgros) [reply packages/synapse-react-client/src/components/AccessRequirementRelatedProjectsList/AccessRequirementRelatedProjectsList.stories.ts:16]: Nice use of component to ensure consistent handling of 403/404 scenarios

**Review by nickgros — APPROVED** [review]
Looks great!

File: packages/synapse-react-client/src/components/AccessRequirementRelatedProjectsList/AccessRequirementRelatedProjectsList.tsx (hunk 4 above)
Comment (hallieswan) [reply packages/synapse-react-client/src/components/AccessRequirementRelatedProjectsList/AccessRequirementRelatedProjectsList.tsx]: Oh great, thanks! Fixed in the next commit

**Pull request merged**